### PR TITLE
feat: add request trial license checkbox to site-init page

### PR DESCRIPTION
**Pull request by KattMingMing**
Resolves https://github.com/sourcegraph/sourcegraph/issues/1123

screenshot 2018-11-27 08 27 08

@francisschmaltz / @sqs feedback on UI / copy?

### CODE REVIEW

**Comment by francisschmaltz** [comment]
Is there somewhere where we describe what the button does?

File: cmd/frontend/graphqlbackend/trial_request.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+// RequestTrial makes a submission to the request trial form.
+func (r *schemaResolver) RequestTrial(ctx context.Context, args *struct {
+	Email string
+}) (*EmptyResponse, error) {
+	email := args.Email
+
+	// If user is authenticated, use their uid and overwrite the optional email field.
+	actor := actor.FromContext(ctx)
+	if actor.IsAuthenticated() {
```
Comment (beyang) [inline cmd/frontend/graphqlbackend/trial_request.go]: Condense to `if actor := actor.FromContext(ctx); actor.IsAuthenticated() {`

Comment (beyang) [inline cmd/frontend/graphqlbackend/trial_request.go]: Condense to `if actor := actor.FromContext(ctx); actor.IsAuthenticated() {`

**Comment by KattMingMing** [comment]
@francisschmaltz - I'll be adding a more visible section to the site-admin configuration page for a user to get their trial license. On this page we could also add a _Learn more_ button that opens to the Enterprise features page. Thoughts?

**Comment by francisschmaltz** [comment]
Having a "learn more" button should suffice. I just want to make sure we explain that we will email them when they click the checkbox. We don't want people to think it's automated.

File: web/src/marketing/backend.tsx [hunk 2]
```diff
@@ -118,3 +118,27 @@ export function fetchSurveyResponseAggregates(): Observable<SurveyResponseConnec
...
+                requestTrial(email: $email) {
+                    alwaysNil
+                }
+            }
+        `,
+        { email }
+    )
+        .pipe(
+            map(dataOrThrowErrors),
+            map(() => undefined)
```
Comment (ijsnow) [inline web/src/marketing/backend.tsx]: Not important but there is the `mapTo` operator for this

File: web/src/site-admin/SiteInitPage.tsx [hunk 3]
```diff
@@ -2,7 +2,7 @@ import * as React from 'react'
 import { Redirect, RouteComponentProps } from 'react-router'
 import * as GQL from '../../../shared/src/graphql/schema'
 import { SignUpArgs, SignUpForm } from '../auth/SignUpPage'
-import { eventLogger } from '../tracking/eventLogger'
```
Comment (ijsnow) [inline web/src/site-admin/SiteInitPage.tsx]: Reason for removing event logging in this PR? Seems like we'd lose that telemetry if they don't check it.

Comment (KattMingMing) [reply web/src/site-admin/SiteInitPage.tsx]: `eventLogger` is only initialized on Sourcegraph.com so this was a no-op.

**Pull request merged**